src/geo/transform.js

```diff
@@ -537,9 +537,18 @@ class Transform {
         // Add a bit extra to avoid precision problems when a fragment's distance is exactly `furthestDistance`
         const farZ = furthestDistance * 1.01;
 
+        // The larger the value of nearZ is
+        // - the more depth precision is available for features (good)
+        // - clipping starts appearing sooner when the camera is close to 3d features (bad)
+        //
+        // Smaller values worked well for mapbox-gl-js but deckgl was encountering precision issues
+        // when rendering it's layers using custom layers. This value was experimentally chosen and
+        // seems to solve z-fighting issues in deckgl while not clipping buildings too close to the camera.
+        const nearZ = this.height / 50;
```

Comment:
LGTM
On native side, [near set to `0.1 * state.getCameraToCenterDistance()`](https://github.com/mapbox/mapbox-gl-native/blob/286bc66ba4eef043af1d67858f8f5ec540717f31/src/mbgl/renderer/paint_parameters.cpp#L26) evaluates to height / 90.

Reply:
Is there an existing render test that exercises this with extrusion layers?